app/controllers/katello/api/v2/systems_controller.rb

```diff
@@ -70,6 +70,12 @@ def index_relation
         collection = System.readable
       end
 
+      if params[:available_for] && params[:available_for] == 'host_collection'
+        system_ids = HostCollection.find(params[:host_collection_id]).systems.pluck(:id)
+        collection = collection.where("id NOT IN (?)", system_ids) unless system_ids.empty?
+        return collection
+      end
```

Comment:
ah no difference, I was trying it out in a function, but then just added to index relation and forgot to delete the function. I am for keeping the logic here in index relation, but I am open to suggestion.